File: backend/IbuClubs.Api/Contracts/DTOs/Activity/CreateActivityDto.cs
using System.ComponentModel.DataAnnotations;

namespace IbuClubs.Api.Contracts.DTOs.Activity;

public class CreateActivityDto
{
    [StringLength(36)]
    public string ClubId { get; set; }
    
    [StringLength(100)]
    public string Name { get; set; }
    
    [StringLength(256)]
    public string Description { get; set; }
    
    public DateOnly Date { get; set; }
    
    public TimeOnly Time { get; set; }
    
    public int ActivityPoints { get; set; }
    
    public int MaxParticipants { get; set; }
}

File: mobile/ibuclubs_mobile/lib/features/activities/data/dto/create_activity_dto.dart
import 'package:built_value/built_value.dart';
import 'package:built_value/serializer.dart';

part 'create_activity_dto.g.dart';

abstract class CreateActivityDto
    implements Built<CreateActivityDto, CreateActivityDtoBuilder> {
  String get clubId;
  String get name;
  String get description;
  String get date;
  String get time;
  int get activityPoints;
  int get maxParticipants;

  factory CreateActivityDto([void Function(CreateActivityDtoBuilder) updates]) =
      _$CreateActivityDto;

  CreateActivityDto._();

  static Serializer<CreateActivityDto> get serializer =>
      _$createActivityDtoSerializer;
}
